qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,612 @@
+
+        self.number = number
+
+    def get_raw(self) -> ParamRawDataType:
+        return self.instrument._get_data(trace_num=self.number)
+
+
+class SpectrumAnalyzerMode(InstrumentChannel):
+    """
+    Spectrum Analyzer Mode for Keysight N9030B instrument.
+    """
+
+    def __init__(self, parent: "N9030B", name: str, *arg: Any, **kwargs: Any):
+        super().__init__(parent, name, *arg, **kwargs)
+
+        self._min_freq = -8e7
+        self._valid_max_freq: Dict[str, float] = {"503": 3.7e9,
+                                                  "508": 8.5e9,
+                                                  "513": 13.8e9,
+                                                  "526": 27e9,
+                                                  "544": 44.5e9}
+        opt: str
+        for hw_opt_for_max_freq in self._valid_max_freq.keys():
+            if hw_opt_for_max_freq in self.root_instrument._options():
+                opt = hw_opt_for_max_freq
+        self._max_freq = self._valid_max_freq[opt]
+
+        self.add_parameter(
+            name="start",
+            unit="Hz",
+            get_cmd=":SENSe:FREQuency:STARt?",
+            set_cmd=self._set_start,
+            get_parser=float,
+            vals=Numbers(self._min_freq, self._max_freq - 10),
+            docstring="start frequency for the sweep"
+        )
+
+        self.add_parameter(
+            name="stop",
+            unit="Hz",
+            get_cmd=":SENSe:FREQuency:STOP?",
+            set_cmd=self._set_stop,
+            get_parser=float,
+            vals=Numbers(self._min_freq + 10, self._max_freq),
+            docstring="stop frequency for the sweep"
+        )
+
+        self.add_parameter(
+            name="center",
+            unit="Hz",
+            get_cmd=":SENSe:FREQuency:CENTer?",
+            set_cmd=self._set_center,
+            get_parser=float,
+            vals=Numbers(self._min_freq + 5, self._max_freq - 5),
+            docstring="Sets and gets center frequency"
+        )
+
+        self.add_parameter(
+            name="span",
+            unit="Hz",
+            get_cmd=":SENSe:FREQuency:SPAN?",
+            set_cmd=self._set_span,
+            get_parser=float,
+            vals=Numbers(10, self._max_freq - self._min_freq),
+            docstring="Changes span of frequency"
+        )
+
+        self.add_parameter(
+            name="npts",
+            get_cmd=":SENSe:SWEep:POINts?",
+            set_cmd=self._set_npts,
+            get_parser=int,
+            vals=Ints(1, 20001),
+            docstring="Number of points for the sweep"
+        )
+
+        self.add_parameter(
+            name="sweep_time",
+            label="Sweep time",
+            get_cmd=":SENSe:SWEep:TIME?",
+            set_cmd=":SENSe:SWEep:TIME {}",
+            get_parser=float,
+            unit="s",
+            docstring="gets sweep time"
+        )
+
+        self.add_parameter(
+            name="auto_sweep_time_enabled",
+            initial_value=False,
```

Comment:
Is it intentional that we set this value on the instrument with initializing the driver? Same question below for `auto_sweep_type_enabled`

Reply:
No, I will remove these :)